mantidproject/mantid
# Workbench: Project Save for Matrix and Table workspace displays

**Description of work.**
Added Two Decoders and Encoders for the Matrix and Table Workspace displays. The windows should be now saved and popped back up as they were previously.

**To test:**
- Open workbench
- Load a workspace that uses MatrixWorkspaces
- Load a workspace that uses TableWorkspaces
- Open both the Matrix and Table Workspace displays
- Save the project
- Open the project and it should open another set of windows.
- Close workbench
-  Open Workbench
- Open the project again
- If all has worked then the code works.



Fixes #24579 . 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
